### PR TITLE
Improve ConfigMap concept

### PR DESCRIPTION
**Pull request by tengqm**
This PR adds some missing information to the ConfigMap concept page.

### CODE REVIEW

File: content/en/docs/concepts/configuration/configmap.md [hunk 1]
```diff
@@ -16,33 +16,45 @@ or use additional (third party) tools to keep your data private.
...
+Kubernetes objects that have a `spec`, a ConfigMap has a `data` field and a
+`binaryData` to store key-value pairs. Both the `data` field and the
+`binaryData` are optional. The `data` field is designed to contain UTF-8
+byte sequences while the `binaryData` field is designed to contain binary
+data.
 
 The name of a ConfigMap must be a valid
 [DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).
 
+Each key under the `data` or the `binaryData` field must consists of
```
Comment (feloy) [inline content/en/docs/concepts/configuration/configmap.md:51]: consist

File: content/en/docs/concepts/configuration/configmap.md [hunk 2]
```diff
@@ -224,12 +242,13 @@ data has the following advantages:
 
 - protects you from accidental (or unwanted) updates that could cause applications outages
 - improves performance of your cluster by significantly reducing load on kube-apiserver, by
-closing watches for config maps marked as immutable.
+  closing watches for ConfigMaps marked as immutable.
+
+This feature is controlled by the `ImmutableEphemeralVolumes`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
+You can create an immutable ConfigMap by setting the `immutable` field to `true`.
+For example,
```
Comment (feloy) [inline content/en/docs/concepts/configuration/configmap.md:250]: Perhaps `:` instead of `,` at the end?

Comment (tengqm) [reply content/en/docs/concepts/configuration/configmap.md]: done.

File: content/en/docs/concepts/configuration/configmap.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/concepts/configuration/configmap.md]: done

File: content/en/docs/concepts/configuration/configmap.md [hunk 3]
```diff
@@ -16,33 +16,45 @@ or use additional (third party) tools to keep your data private.
...
-
+You write the code to look in an environment variable named `DATABASE_HOST`.
+Locally, you set that variable to `localhost`. In the cloud, you set it to
+refer to a Kubernetes {{< glossary_tooltip text="Service" term_id="service" >}}
+that exposes the database component to your cluster.
 This lets you fetch a container image running in the cloud and
 debug the exact same code locally if needed.
 
+A ConfigMap is not designed to hold large chunks of data. The data stored in a
+ConfigMap object cannot exeed 1 MiB. If you need to store settings that are
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/configmap.md:34]: nit: can remove "object" after ConfigMap?
nit: could drop the clause, "for example" at end of line 36.

File: content/en/docs/concepts/configuration/configmap.md [hunk 4]
```diff
@@ -16,33 +16,45 @@ or use additional (third party) tools to keep your data private.
...
+separate database or file service, for example.
+
 ## ConfigMap object
 
 A ConfigMap is an API [object](/docs/concepts/overview/working-with-objects/kubernetes-objects/)
 that lets you store configuration for other objects to use. Unlike most
-Kubernetes objects that have a `spec`, a ConfigMap has a `data` section to
-store items (keys) and their values.
+Kubernetes objects that have a `spec`, a ConfigMap has a `data` field and a
+`binaryData` to store key-value pairs. Both the `data` field and the
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/configmap.md:43]: Possibly edit: Unlike most ... a ConfigMap has `data` and `binaryData` fields. These fields are optional
and store data as key-value pairs.

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-23807--kubernetes-io-master-staging.netlify.app/docs/concepts/configuration/configmap/

File: content/en/docs/concepts/configuration/configmap.md [hunk 5]
```diff
@@ -16,33 +16,45 @@ or use additional (third party) tools to keep your data private.
...
+separate database or file service.
+
 ## ConfigMap object
 
 A ConfigMap is an API [object](/docs/concepts/overview/working-with-objects/kubernetes-objects/)
 that lets you store configuration for other objects to use. Unlike most
-Kubernetes objects that have a `spec`, a ConfigMap has a `data` section to
-store items (keys) and their values.
+Kubernetes objects that have a `spec`, a ConfigMap has `data` field and
+`binaryData` fields. These fields accepts key-value pairs as their values.
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/configmap.md:43]: nit: should it be:
a ConfigMap has `data` and `binaryData` fields?

Comment (sftim) [reply content/en/docs/concepts/configuration/configmap.md:43]: That suggested wording looks better to me.

**Review by sftim — COMMENTED** [review]
Thanks @tengqm 
/lgtm

File: content/en/docs/concepts/configuration/configmap.md (hunk 5 above)
Comment (tengqm) [reply content/en/docs/concepts/configuration/configmap.md:43]: yes, will revise.

**Pull request merged**